src/util-cpu.c

```diff
@@ -75,9 +76,15 @@ uint16_t UtilCpuGetNumProcessorsConfigured(void)
 
     return (uint16_t)nprocs;
 #elif OS_WIN32
-	long nprocs = -1;
-	const char* envvar = getenv("NUMBER_OF_PROCESSORS");
-	nprocs = (NULL != envvar) ? atoi(envvar) : 0;
+    int64_t nprocs = 0;
+    const char* envvar = getenv("NUMBER_OF_PROCESSORS");
+    if (envvar != NULL) {
+        if (StringParseInt64(&nprocs, 10, 0, envvar) < 0) {
+            SCLogWarning(SC_ERR_INVALID_VALUE, "Invalid value for number of "
```

Comment:
Suggest using `SC_WARN` code?

Reply:
Does `SC_WARN_POOR_RULE` fit the category here? I do not see any appropriate SC_WARN codes for invalid value.

Reply:
I don't object to the original code you picked. We want to overhaul the error codes anyway.